pkg/trace/stats/otel_util_test.go

```diff
@@ -211,10 +236,13 @@ func TestProcessOTLPTraces(t *testing.T) {
 			conf.Hostname = agentHost
 			conf.DefaultEnv = agentEnv
 			conf.Features["enable_cid_stats"] = struct{}{}
+			if tt.enableReceiveResourceSpansV2 {
+				conf.Features["enable_receive_resource_spans_v2"] = struct{}{}
+			}
 			conf.PeerTagsAggregation = tt.peerTagsAggr
 			conf.OTLPReceiver.AttributesTranslator = attributesTranslator
 			conf.OTLPReceiver.SpanNameAsResourceName = tt.spanNameAsResourceName
-			if conf.OTLPReceiver.SpanNameAsResourceName {
+			if conf.OTLPReceiver.SpanNameAsResourceName || tt.enableOperationAndResourceNameV2 {
```

Comment:
```suggestion
			if tt.enableOperationAndResourceNameV2 {
```

why does SpanNameAsResourceName enable OperationAndResourceNameV2? This seems a bug before

Reply:
the test appears to be written to ensure that even if OperationAndResourceNameV2 is enabled that SpanNameAsResourceName logic overrides it. I don't love the setup either but don't think its a bug in the test